buildMozillaMach: fix builds with crash reporting disabled

squalus opened the pull request:
###### Description of changes
- only try to build and copy crash reporting symbols when crash reporting is enabled
- fixes the librewolf build

Fixes #173129

<!--
For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

greydot commented:
Yup! This just about fixes it. God bless my threadripper for fast builds :D

squalus commented:
The Librewolf build succeeds and passes basic browsing tests. cc @mweinelt

mweinelt commented on pkgs/applications/networking/browsers/firefox/common.nix:
```diff
@@ -169,10 +169,8 @@ buildStdenv.mkDerivation ({
-    "out"
-    "symbols"
-  ];
+  outputs = [ "out" ]
```
```suggestion
  outputs = [
    "out"
  ]
```

To match the list style everywhere else.

squalus commented:
Firefox build also succeeds and passes basic browsing tests.

mweinelt commented on pkgs/applications/networking/browsers/firefox/common.nix:
```diff
@@ -217,6 +214,7 @@ buildStdenv.mkDerivation ({
     wrapGAppsHook
   ]
   ++ lib.optionals pgoSupport [ xvfb-run ]
+  ++ lib.optionals crashreporterSupport [ dump_syms ]
```
```suggestion
  ++ lib.optionals crashreporterSupport [ dump_syms ]
  ++ lib.optionals pgoSupport [ xvfb-run ]
```

Alphabetical ordering

mweinelt reviewed (changes requested):
Two nits and sorry for breaking librewolf.

squalus committed:
buildMozillaMach: fix builds with crash reporting disabled

- only try to build and copy crash reporting symbols when crash
  reporting is enabled
- fixes the librewolf build

squalus commented:
Applied the suggested changes.
